src/company_base.h

```diff
@@ -54,11 +55,11 @@ extern CompanyPool _company_pool;
 struct CompanyProperties {
 	uint32 name_2;                   ///< Parameter of #name_1.
 	StringID name_1;                 ///< Name of the company if the user did not change it.
-	char *name;                      ///< Name of the company if the user changed it.
+	std::string name;                ///< Name of the company if the user changed it.
```

Comment:
Empty string if not user set?

Reply:
Yeah, I've checked and no function would actually allow setting an empty name (applies to the other user defined names as well).  I can add it to the line comment if you prefer.